### PR TITLE
forbid parse apis

### LINKED ISSUE #2472: Add forbidden API for number parsing methods
Add a custom Forbidden API check for `Integer.parse()`, `Long.parse()` that suggests using the `ParseUtil` equivalent.

### PR DESCRIPTION
**Pull request by Taz03**
fixes #2472

### CODE REVIEW

File: oshi-core/src/main/java/oshi/hardware/platform/unix/freebsd/FreeBsdCentralProcessor.java [hunk 1]
```diff
@@ -207,15 +207,16 @@ private static List<LogicalProcessor> parseTopology() {
 
                     // Regex guarantees parsing digits so we won't get a
                     // NumberFormatException
+                    long parsedVal = ParseUtil.hexStringToLong(firstVal, 0);
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/platform/unix/freebsd/FreeBsdCentralProcessor.java:210]: Interesting.... the comments indicate the regex guarantees the parsing won't fail.  But at the same time it's more readable with the hex string parsing. But then when you look at that method it's a lot of manual manipulation when I think the actual parsing-with-radix would handle it!

Anyway, probably ok here.

File: oshi-core/src/main/java/oshi/util/EdidUtil.java [hunk 2]
```diff
@@ -32,6 +33,7 @@ private EdidUtil() {
      * @param edid The EDID byte array
      * @return The manufacturer ID
      */
+    @SuppressForbidden(reason = "using base 2")
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/util/EdidUtil.java:36]:
```suggestion
    @SuppressForbidden(reason = "customized base 2 parsing not in Util class")
```
I think we need to catch a NumberFormatException here?  Do you agree?

**Review by dbwiddis — COMMENTED** [review]
Initial pass...

File: oshi-core/src/main/java/oshi/util/EdidUtil.java (hunk 2 above)
Comment (Taz03) [reply oshi-core/src/main/java/oshi/util/EdidUtil.java]: if we're doing it in other similar places then, yes. Also its out of the scope of this pr I can open a new pr if u want to catch the exception

File: config/forbidden-apis.txt [hunk 3]
```diff
@@ -1,5 +1,9 @@
 java.lang.Integer#parseInt(java.lang.String) @ Use oshi.util.ParseUtil#parseIntOrDefault(java.lang.String,int)
+java.lang.Integer#parseInt(java.lang.String,int)
+java.lang.Integer#parseInt(java.lang.CharSequence,int,int,int)
```
Comment (dbwiddis) [inline config/forbidden-apis.txt:3]: See [docs](https://jenkins.thetaphi.de/job/Forbidden-APIs/javadoc/signatures-syntax.html) for syntax of multi-line reasons.
```suggestion
@defaultMessage Use existing oshi.util.ParseUtil methods or create a new one

java.lang.Integer#parseInt(java.lang.String)
java.lang.Integer#parseInt(java.lang.String,int)
java.lang.Integer#parseInt(java.lang.CharSequence,int,int,int)
```

**Review by dbwiddis — COMMENTED** [review]
Just a suggestion for a single combined message for all these parse methods. 

Also were we going to add `Integer#decode(String)` and `Long#decode(String)`?

**Comment by Taz03** [comment]
> Also were we going to add `Integer#decode(String)` and `Long#decode(String)`?

no?

**Pull request merged**